source/shared/cpp/ObjectModel/ParseUtil.h

```diff
@@ -148,15 +145,7 @@ namespace AdaptiveSharedNamespace
             auto const& propertyValue = json.get(propertyName, Json::Value());
             if (propertyValue.empty())
             {
-                if (isRequired)
-                {
-                    throw AdaptiveCardParseException(ErrorStatusCode::RequiredPropertyMissing,
-                                                     "Property is required but was found empty: " + propertyName);
-                }
-                else
-                {
-                    return defaultEnumValue;
-                }
+                return std::nullopt;
```

Comment:
@RebeccaAnne,
Returnnig nullopt for empty value seems to be inconsistent with returnning the default value from the line 157 when the given string value doesn't map to enum values. For example, for "", will return nullopt, but "Hello World" for horizontal alignment will return HorizontalAlignmentLeft. returnning nullopt makes more sense since we have to know if the user set the value or not. But then the renderers have to know what is the proper default values. We could have a util function for returnning the default values for cross platforms.

Reply:
You're first point is a good one, and I updated the enum magic code to not set the default from something like "Hello World". Since most of the callers of this are coming through ParseUtil::GetEnumValue those callers will set the default anyway, but for the ones coming directly through ParseUtil::GetOptionalEnumValue should now get nullopt in that case. 

It is correct that with this setup the renderers do need to know the defaults. I'm not opposed to setting something up in shared model to ensure we've got consistent defaults cross platform. Could you log an issue to track adding that?

Reply:
+1 to this, would love to move the defaults for all optionals back into the object model